src/main/java/com/hotels/beans/utils/ClassUtils.java

```diff
@@ -350,49 +348,92 @@ public int getTotalFields(final Class<?> clazz, final Predicate<? super Field> p
     }
 
     /**
-     * Checks if the destination class uses the Builder pattern.
-     * @param constructor the all args constructor
-     * @param targetClass the destination object class
-     * @param <K> the target object type
-     * @return true if the target class uses the builder pattern
+     * Retrieves all classes defined into the given one.
+     * @param clazz class where we search for a nested class
+     * @return all classes defined into the given one
      */
-    public <K> boolean usesBuilderPattern(final Constructor constructor, final Class<K> targetClass) {
-        final String cacheKey = "UsesBuilderPattern-" + constructor.getDeclaringClass().getName();
-        return cacheManager.getFromCache(cacheKey, Boolean.class).orElseGet(() -> {
-            final boolean res = !isPublic(constructor.getModifiers()) && isNotEmpty(targetClass.getDeclaredClasses());
-            cacheManager.cacheObject(cacheKey, res);
-            return res;
+    public Class[] getDeclaredClasses(final Class<?> clazz) {
+        notNull(clazz, CLAZZ_CANNOT_BE_NULL);
+        String cacheKey = "DeclaredClasses-" + clazz.getName();
+        return cacheManager.getFromCache(cacheKey, Class[].class).orElseGet(() -> {
+            Class[] declaredClasses = clazz.getDeclaredClasses();
+            cacheManager.cacheObject(cacheKey, declaredClasses);
+            return declaredClasses;
         });
     }
 
     /**
-     * Retrieves the all args constructor.
+     * Creates an instance of the given class invoking the no args constructor.
+     * @param objectClass the class of the object to return.
+     * @param <T> the class object type.
+     * @return the object instance.
+     * @throws InvalidBeanException in case the object creation fails.
+     */
+    public <T> T getInstance(final Class<? extends T> objectClass) {
```

Comment:
this method is not used, I think it can be removed

Reply:
Ok removed, I have check that unit test work fine